lib/rbs/collection/config/lockfile.rb

```diff
@@ -0,0 +1,123 @@
+# frozen_string_literal: true
+
+module RBS
+  module Collection
+    class Config
+      class Lockfile
+        attr_reader :lockfile_path, :lockfile_dir, :path, :gemfile_lock_path, :sources, :gems
+
+        def initialize(lockfile_path:, path:, gemfile_lock_path:)
+          @lockfile_path = lockfile_path
+          @lockfile_dir = lockfile_path.parent
+          @path = path
+          @gemfile_lock_path = gemfile_lock_path
+
+          @sources = {}
+          @gems = {}
+        end
+
+        def fullpath
+          lockfile_dir + path
+        end
+
+        def gemfile_lock_fullpath
+          if gemfile_lock_path
+            lockfile_dir + gemfile_lock_path
+          end
+        end
+
+        def each_source(&block)
+          if block
+            sources.each_value(&block)
+            yield Sources::Rubygems.instance
+            yield Sources::Stdlib.instance
+          else
+            enum_for :each_source
+          end
+        end
+
+        def to_lockfile
+          # @type var data: lockfile_data
+
+          data = {
+            "sources" => sources.each_value.sort_by {|s| s.name }.map do |source|
+              # @type block: git_source_data
+              {
+                "name" => source.name,
+                "remote" => source.remote,
+                "revision" => source.resolved_revision,
+                "repo_dir" => source.repo_dir
+              }
```

Comment:
Is this replaceable with `Source#to_lockfile`?

Reply:
Maybe? But, the `Source#to_lockfile` includes `"type"` while existing lockfiles don't.

Reply:
Hmm, I think it's not a problem because it's not a breaking change. Including `type` is more descriptive, and we can keep it more DRY.

Reply:
We will have `type` attribute. We are planning to add another `type` of sources soon. So having `git` type here is nice.